dips/dip-158.md

```diff
@@ -0,0 +1,795 @@
+| timestamp | uint | Y | Unix timestamp indicating the time that the payment Command was created.
+|
+
+```
+{
+    "amount": 100000000,
+    "currency": "XUS",
+    "action": "auth",
+    "valid_until": 74000,
+    "timestamp": 72322,
+}
+```
+
+## Additional Command Types
+The following is a list of new values for the `command_type` field of the `CommandRequestObject`. Each command type denotes a step in the off-chain negotiations and specifies the data of the requests and the response 
+* `InfoCommand` - Allows obtianing payment information based on reference id
+* `InitCommand` - Initializes the payment process
+* `ReadyCommand` - Notifies the payment is ready for settlement on-chain
+* `AuthorizedCommand` - Notifies that payment funds are authorized (locked) for future capture
+* `CaptureCommand` - Request to capture previously authorized (locked) funds 
+* `VoidCommand` - Voids a payment
+* `AbortCommand` - Aborts a payment
+
+Accordingly, the following list specifies additional values for `_ObjectType` field of  `CommandObject`
+* `InfoCommand` 
+* `InitCommand` 
+* `ReadyCommand`
+* `AuthorizedCommand` 
+* `CaptureCommand`
+* `VoidCommand`
+* `AbortCommand`
+
+## PaymentInfoObject (new)
+The minimal payment details required in order for the payer to decide whether the accept the payment or reject it. 
+
+Field|Type|Required?|Description|
+|-|-|-|-|
+|receiver|PaymentActorObject|Y|Only the address field of the PaymentActorObject is required. i.e. status element is not required|
+|action|PaymentActionObject|Y|Information regarding the type of payment and payment amount|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|description|str|N|Description of the payment. To be displayed to the user|
+|
+
+```
+"payment_info": 
+{
+  "receiver": {
+    "address": "dm1pllhdmn9m42vcsamx24zrxgs3qqqpzg3ng32kvacapx5fl",         
+    "merchant_data": { 
+      "name": "Acme",
+      "legal_name": "Acme Technologies Inc.",
+      "address": {
+        "city": "San Francisco",
+        "country": "US",
+        "line1": "1260 Market Street",
+        "line2": "Suite 450",
+        "postal_code": "94103",
+        "state": "CA"
+      }        
+    }        
+  },
+  "action": {
+    "amount": "100000000",
+    "currency": "XUS",
+    "action": "charge",
+    "timestamp": "72322"
+  },
+  "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+  "description": "payment request description"     
+}
+```
+
+## MerchantDataObject (new)
+The details of the merchant.
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|name|str|Y|Merchant's display name. Should be recognizable by the payer|
+|legal_name|str|Y|The legal entity name|
+|address|AddressObject|Y|Merchant's physical address information|
+|
+```
+"merchant_data": 
+{
+  "name": "Acme",
+  "legal_name": "Acme Technologies Inc.",
+  "address": {
+    AddressObject()
+  }        
+}
+``` 
+
+## PayerDataObject (new)
```

Comment:
Customer?

Reply:
I think in this case payer is more accurate as there could be some scenarios where the customer is not the payer.